sdk/resource_locator.go

```diff
@@ -54,7 +65,7 @@ func NewResourceLocatorFromReader(reader io.Reader) (*ResourceLocator, error) {
 	if err != nil {
 		return rl, err
 	}
-	rl.protocol = urlProtocol(oneByte[0])
+	rl.protocol = protocolHeader(oneByte[0])
 
 	_, err = reader.Read(oneByte)
```

Comment:
You don't seem to be reading the identifier here?

You could do something like:

```
identifier := make([]byte, rl.protocl.identifierLength)
reader.Read(identifier)
```

Reply:
yep, added with unit tests.  The SDK sets the Key Identifier, then KAS reads it.  Added to SDK for completeness